File: day4/part1.py
#!/usr/bin/env python3

valid_phrase_count = 0
with open('input.txt') as file:
    for line in file.readlines():
        words = line.strip().split()
        if len(set(words)) == len(words):
            valid_phrase_count += 1

print(valid_phrase_count)

File: 2021/day1/part1.ts
#!/usr/bin/env npx ts-node

import * as fs from 'fs';
import { exit } from 'process';

const measurements = fs
    .readFileSync('./input.txt', 'utf-8')
    .split(/\n/)
    .filter(line => line != '')
    .map(numStr => Number(numStr));

let increments = 0;
let prev = measurements.shift()!;

measurements.forEach(current =>  {
    if (current > prev) {
        increments += 1;
    }
    prev = current;
});

console.log(increments);